sky/provision/aws/instance.py

```diff
@@ -791,3 +793,19 @@ def get_cluster_info(region: str,
         instances=instances,
         head_instance_id=head_instance_id,
     )
+
+
+def query_ports(
+    cluster_name_on_cloud: str,
+    ip: str,
+    ports: List[str],
+    provider_config: Optional[Dict[str, Any]] = None,
+) -> Dict[int, List[common.Endpoint]]:
+    """See sky/provision/__init__.py"""
+    del cluster_name_on_cloud, provider_config  # Unused.
+    ports = list(resources_utils.port_ranges_to_set(ports))
+    result: Dict[int, List[common.Endpoint]] = {}
+    for port in ports:
+        result[port] = [common.SocketEndpoint(host=ip, port=port)]
```

Comment:
It seems like if we opened `30000-31000`, this will return a dict with 1000 entries. It will probably overflood the terminal... Is there a better way top handle this?

Reply:
Yes this is an outstanding issue, will think about it.